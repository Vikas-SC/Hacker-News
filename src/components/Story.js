import React, { useEffect, useState } from 'react';

import { getAllStories } from '../services/hnApi';

import { localStore } from '../utils/localStore';

import '../styles/style.scss';

export const Story = () => {
  const [stories, setStories] = useState([]);
  const [pageNumber, setPageNumber] = useState(0);
  const [disable, setDisable] = useState(true);

  useEffect(() => {
    const storiesFromLocalStorage = localStore.getNews();
    if (!storiesFromLocalStorage) {
      const res = getAllStories(pageNumber).then(stories => stories);
      res.then((data) => {
        const hits = data.data.hits;
        localStore.setNews(hits);
        setStories(hits);
      });
    } else {
      setStories(storiesFromLocalStorage);
    }
    if (pageNumber < 1) {
      setDisable(true);
    } else {
      setDisable(false)
    }
  }, [pageNumber, disable]);

  const nextNews = () => {
    localStorage.clear();
    setPageNumber(pageNumber + 1);
  }

  const prevNews = () => {
    localStorage.clear();
    setPageNumber(pageNumber - 1);
  }

  const upVote = (event, story) => {
    event.stopPropagation();
    const storedStories = localStore.getNews();
    const updatedStories = storedStories.map((upVoteStory) => {
      if (upVoteStory.objectID === story.objectID) {
        upVoteStory.points += 1;
      }
      return upVoteStory;
    });
    setStories(updatedStories);
    localStore.setNews(updatedStories);
  }
  return (
    <div className="container">
      <table className="table table-striped" id="newsTable">
        <thead className="tableHead">
          <tr>
            <th>Comments</th>
            <th>Vote Count</th>
            <th>Up Vote</th>
            <th>News Details</th>
          </tr>
        </thead>

        <tbody>
          {stories ? stories.map((story) => {
            return (<tr key={story.objectID} onClick={(e) => upVote(e, story)}>
              <td>{story.num_comments || '--'}</td>
              <td>{story.points || '--'}</td>
              <td> <span className="glyphicon glyphicon-triangle-top" value={story.points}></span></td>
              <td>{story.title || '--'}</td>
            </tr>)
          }) :
            <div className="d-flex justify-content-center">
              <div className="spinner-border" role="status">
                <span classN="sr-only">Loading...</span>
              </div>
            </div>

          }
        </tbody>

      </table>
      <div className="paginationFloat">
        <nav aria-label="Page navigation example">
          <ul className="pagination justify-content-end">
            <li className={`page-item ${disable ? 'disabled' : ''}`}>
              <a className="page-link" onClick={() => prevNews()}>Previous</a>
            </li>
            <li className="page-item">
              <span className="page-link" onClick={() => nextNews()}>Next</span>
            </li>
          </ul>
        </nav>
      </div>
    </div >
  )
}