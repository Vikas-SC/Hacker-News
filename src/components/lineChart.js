import React, { Component } from "react";
import Chart from "react-apexcharts";

class LineChart extends Component {
  constructor(props) {
    super(props);
    const ids = props.stories.map(id => id.objectID);
    const votes = props.stories.map(point => point.points);
    this.state = {
      options: {
        chart: {
          id: "basic-line",
          toolbar: {
            show: false
          },
        },
        xaxis: {
          categories: ids,
          title: {
            text: 'ID',
          }
        },
        yaxis: {
          title: {
            text: 'VOTE',
          },
        }
      },
      series: [
        {
          name: "series-1",
          data: votes
        }
      ]
    };
  }

  render() {
    return (
      <div className="app">
        <div className="row">
          <div className="mixed-chart">
            <Chart
              options={this.state.options}
              series={this.state.series}
              type="line"
              width="100%"
            />
          </div>
        </div>
      </div>
    );
  }
}

export default LineChart;