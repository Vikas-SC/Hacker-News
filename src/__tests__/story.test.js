import React from 'react';
import { Story } from '../components/Story';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Story></Story>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});