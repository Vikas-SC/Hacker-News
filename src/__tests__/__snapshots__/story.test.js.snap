// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <table
    className="table table-striped"
    id="newsTable"
  >
    <thead
      className="tableHead"
    >
      <tr>
        <th>
          Comments
        </th>
        <th>
          Vote Count
        </th>
        <th>
          Up Vote
        </th>
        <th>
          News Details
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <div
    className="paginationFloat"
  >
    <nav
      aria-label="Page navigation example"
    >
      <ul
        className="pagination justify-content-end"
      >
        <li
          className="page-item disabled"
        >
          <a
            className="page-link"
            onClick={[Function]}
          >
            Previous
          </a>
        </li>
        <li
          className="page-item"
        >
          <span
            className="page-link"
            onClick={[Function]}
          >
            Next
          </span>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
